wagtailio/navigation/models.py

```diff
@@ -75,3 +88,12 @@ class NavigationSettings(BaseSiteSetting, ClusterableModel):
         FieldPanel("main_navigation"),
         FieldPanel("footer_navigation"),
     ]
+
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        keys = [
+            make_template_fragment_key(key, vary_on=[self.site.pk])
+            for key in ["primarynav", "footernav"]
```

Comment:
Issue: This also needs to vary on `request.is_preview`

```suggestion
            make_template_fragment_key(key, vary_on=[self.site.pk, True, False])
            for key in ["primarynav", "footernav"]
```

Reply:
should be `[self.site.pk, False, False]` in that we want to clear the non-preview fragment